### ISSUE TITLE
Training gets slower with each repetition

### ISSUE THREAD

**Issue by macwilam** [fchollet/keras#3579]
Hi,
I'm not attaching any code as it is not specific to any form of my model. Im running keras with tensorflow, everything is up to date.
When I run keras inside pytho notebook (python 3) and try to train a model multiple time to search optimal parameters it gets slower with each iteration. For example from 3 min per iteration to 6 min in 40 iterations. I run a function that builds new model = sequential() in each iteration. 
I looked for any command to "restart" tensorflow or keras but couldn't find anything. Is it a bug or am I doing something wrong?

Kind regards.

**Comment by fchollet** [comment]
If you are creating many TF models in a row, at some point you will need to
clear your TF graph, via:

```
from keras import backend as K
K.clear_session()
```

**Comment by macwilam** [comment]
Thank you very much for your answer. Before posting i looked through all functions in here: https://keras.io/backend/ and didn't find this function.

Thanks again.

**Comment by mfurlend** [comment]
Any progress on this?

**Comment by daroodar** [comment]
> Any progress on this?

@mfurlend what I did was to avoid initiating a new object each time a new model was to be trained because each time a new object is made, a new graph is made on the back end and graph memory keeps on increasing every time you do `Sequential()`.

I achieved this by the following methodology.
Say I had to train 100 models. What I would do is 
- instantiate 1 object of keras.models.Sequential
- store initial weights in an object `initial_weights`
- do a loop for 100 models. In each loop, start with training for that model, store weights at the end of training in a file (also could store in an object) for e.g. as `model_1_weights.h5` and then after storing, reset the weights to `initial_weights` so for the next loop, training will commence from scratch.

This method has the advantage that it uses only 1 tensor flow graph in the back end. This solved the problem for me and training time was approximately the same for each model.
